Sources/NIOPosix/NonBlockingFileIO.swift

```diff
@@ -627,4 +627,132 @@ public struct NonBlockingFileIO: Sendable {
+    ///
+    /// - parameters:
+    ///     - path: The path of the file to get information about.
+    ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
+    /// - returns: An `EventLoopFuture` containing file information.
+    public func lstat(path: String, eventLoop: EventLoop) -> EventLoopFuture<stat> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            var s = stat()
+            try Posix.lstat(pathname: path, outStat: &s)
+            return s
+        }
+    }
+
+    /// Creates a symbolic link to a  `destination` file  at `path` on a private thread pool which is separate from any `EventLoop` thread.
+    ///
+    /// - parameters:
+    ///     - path: The path of the link.
+    ///     - destination: Target path where this link will point to.
+    ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
+    /// - returns: An `EventLoopFuture` which is fulfilled if the rename was successful or fails on error.
+    public func symlink(path: String, to destination: String, eventLoop: EventLoop) -> EventLoopFuture<Void> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            try Posix.symlink(pathname: path, destination: destination)
+        }
+    }
+
+    /// Returns target of the symbolic link at `path` on a private thread pool which is separate from any `EventLoop` thread.
+    ///
+    /// - parameters:
+    ///     - path: The path of the link to read.
+    ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
+    /// - returns: An `EventLoopFuture` containing link target.
+    public func readlink(path: String, eventLoop: EventLoop) -> EventLoopFuture<String> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            let maxLength = Int(PATH_MAX)
+            let pointer = UnsafeMutableBufferPointer<CChar>.allocate(capacity: maxLength)
+            defer {
+                pointer.deallocate()
+            }
+            let length = try Posix.readlink(pathname: path, outPath: pointer.baseAddress!, outPathSize: maxLength)
+            return String(decoding: UnsafeRawBufferPointer(pointer).prefix(length), as: UTF8.self)
+        }
+    }
+
+    /// Removes symbolic link at `path` on a private thread pool which is separate from any `EventLoop` thread.
+    ///
+    /// - parameters:
+    ///     - path: The path of the link to remove.
+    ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
+    /// - returns: An `EventLoopFuture` which is fulfilled if the rename was successful or fails on error.
+    public func unlink(path: String, eventLoop: EventLoop) -> EventLoopFuture<Void> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            try Posix.unlink(pathname: path)
+        }
+    }
+
+    public struct DirectoryEntry {
```

Comment:
This type needs a doc comment, both on the type and on the properties. All the properties need to be public, we need a public memberwise initializer, and this should probably be `Hashable` too.

Reply:
Should this be namespaced or a top-level type? wdyt?

Reply:
added doc comment, hashable conformance and public memberwise init, made fields public and moved outside of `NonBlockingFileIO` type for now